#
# This file and its contents are supplied under the terms of the
# Common Development and Distribution License ("CDDL)". You may
# only use this file in accordance with the terms of the CDDL.
#
# A full copy of the text of the CDDL should have accompanied this
# source. A copy of the CDDL is also available via the Internet at
# http://www.illumos.org/license/CDDL.
#

#
# Copyright 2011 EveryCity Ltd. All rights reserved.
#

# This patch fixes the following error on smartos:

# make[3]: Entering directory `/export/home/alasdair/s10-userland/components/postgresql91/build/i86/src/port'
# /ec/bin/gcc -m32 -O3  -I/ec/include -I/ec/include -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wformat-security -fno-strict-aliasing -fwrapv -I../../src/port -DFRONTEND -I../../src/include -I/export/home/alasdair/s10-userland/components/postgresql91/postgresql-9.1.1/src/include -I/ec/include/libxml2   -c -o getpeereid.o /export/home/alasdair/s10-userland/components/postgresql91/postgresql-9.1.1/src/port/getpeereid.c
#In file included from /usr/include/sys/procfs.h:50,
#                 from /usr/include/sys/ucred.h:33,
#                 from /export/home/alasdair/s10-userland/components/postgresql91/postgresql-9.1.1/src/port/getpeereid.c:27:
# /usr/include/sys/old_procfs.h:52:2: error: #error "Cannot use procfs in the large file compilation environment"

--- postgresql-9.2.2/configure.in.~1~	2011-09-22 21:57:57.000000000 +0000
+++ postgresql-9.2.2/configure.in	2011-12-28 01:27:12.999480266 +0000
@@ -19,10 +19,6 @@
 
 AC_INIT([PostgreSQL], [9.2.2], [pgsql-bugs@postgresql.org])
 
-m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.63], [], [m4_fatal([Autoconf version 2.63 is required.
-Untested combinations of 'autoconf' and PostgreSQL versions are not
-recommended.  You can remove the check from 'configure.in' but it is then
-your responsibility whether the result works or not.])])
 AC_COPYRIGHT([Copyright (c) 1996-2012, PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
@@ -969,7 +965,7 @@
 ##
 
 dnl sys/socket.h is required by AC_FUNC_ACCEPT_ARGTYPES
-AC_CHECK_HEADERS([crypt.h dld.h fp_class.h getopt.h ieeefp.h ifaddrs.h langinfo.h poll.h pwd.h sys/ioctl.h sys/ipc.h sys/poll.h sys/pstat.h sys/resource.h sys/select.h sys/sem.h sys/shm.h sys/socket.h sys/sockio.h sys/tas.h sys/time.h sys/ucred.h sys/un.h termios.h ucred.h utime.h wchar.h wctype.h])
+AC_CHECK_HEADERS([crypt.h dld.h fp_class.h getopt.h ieeefp.h ifaddrs.h langinfo.h poll.h pwd.h sys/ioctl.h sys/ipc.h sys/poll.h sys/pstat.h sys/resource.h sys/select.h sys/sem.h sys/shm.h sys/socket.h sys/sockio.h sys/tas.h sys/time.h sys/un.h termios.h ucred.h utime.h wchar.h wctype.h])
 
 # On BSD, cpp test for net/if.h will fail unless sys/socket.h
 # is included first.
